airbyte-lib/airbyte_lib/_factories/cache_factories.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+
+from pathlib import Path
+
+import ulid
+
+from airbyte_lib.caches.duckdb import DuckDBCache, DuckDBCacheConfig
+
+
+def get_default_cache() -> DuckDBCache:
+    """Get a local cache for storing data, using the default database path.
+
+    Cache files are stored in the `.cache` directory, relative to the current
+    working directory.
+    """
+    config = DuckDBCacheConfig(
+        db_path="./.cache/default_cache_db.duckdb",
+    )
+    return DuckDBCache(config=config)
+
+
+def new_local_cache(
+    cache_name: str | None = None,
+    cache_dir: str | Path | None = None,
+    cleanup: bool = True,
+) -> DuckDBCache:
+    """Get a local cache for storing data, using a name string to seed the path.
+
+    Args:
+        cache_name: Name to use for the cache. Defaults to None.
+        root_dir: Root directory to store the cache in. Defaults to None.
+        cleanup: Whether to clean up temporary files. Defaults to True.
+
+    Cache files are stored in the `.cache` directory, relative to the current
+    working directory.
+    """
+    if cache_name:
+        if " " in cache_name:
+            raise ValueError(f"Cache name '{cache_name}' cannot contain spaces")
+
+        if not cache_name.isalnum():
+            raise ValueError(f"Cache name '{cache_name}' can only contain alphanumeric characters")
+
+    cache_name = cache_name or str(ulid.ULID())
+    cache_dir = cache_dir or Path("./.cache/{cache_name}")
+    if not isinstance(cache_dir, Path):
+        cache_dir = Path(cache_dir)
+
+    config = DuckDBCacheConfig(
+        db_path=cache_dir / f"db_{cache_name}.duckdb",
```

Comment:
This will produce files like `.cache/01HKYK4KDF0653863VQ79MRH43/db_01HKYK4KDF0653863VQ79MRH43.duckdb`

The repeated ulid is a little weird, but not a big deal - no need to change, just wanted to mention I stumbled over it a bit.

Reply:
Yeah. I think this is necessary for now, just for isolation of each cache iteration. If they give a name, or if they use the default cache, it'll be a shorter name.

Maybe in a future pass, we can shorten the string to the first 10 chars or so, or some similar decision.

One factor here is that DuckDB usees the file name as the DB name. So, having the name repeated is pretty much required if we want DB name to match cache name. And the dedicated subfolder is because we want to keep all cache artifacts next to each other.

There are probably other good ways to organize these, but for now I think it works for what we need. Happy to discuss a next iteration that changes things up.